components/pcloud/actions/create-folder/create-folder.js

```diff
@@ -0,0 +1,66 @@
+const common = require("../../common");
+const { pcloud } = common.props;
+const validate = require("validate.js");
+
+module.exports = {
+  key: "pcloud-create-folder",
+  name: "Create Folder",
+  description: "Creates a folder in the specified path or folder.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    pcloud,
+    domainLocation: {
+      propDefinition: [
+        pcloud,
+        "domainLocation",
+      ],
+    },
+    folderId: {
+      propDefinition: [
+        pcloud,
+        "folderId",
+      ],
+      description: "Id of the parent folder where the new folder will be created.",
+    },
+    path: {
+      propDefinition: [
+        pcloud,
+        "path",
+      ],
+      description: "Path to the parent folder, where the new folder will be created.",
+    },
+    name: {
+      propDefinition: [
+        pcloud,
+        "name",
+      ],
+    },
+  },
+  methods: {
+    ...common.methods,
+  },
+  async run() {
+    const constraints = {
```

Comment:
I was just going to delete that presence validation, but here is this scenario.

it turns out that required parameter can be bypassed when a user sets the required parameter to a reference such as a variable from previous steps. in this way, the workflow won't complain about the required parameter because it holds a value (a reference to the previous steps), but it won't internally check that this reference is `null`, or `undefined`.

Here in the screenshot is what I mean:

![image](https://user-images.githubusercontent.com/25184117/131620280-70b58490-b818-463b-874a-552f4023ee99.png)

and to avoid sending unset values in the required props, is why presence is being checked.
when an required parameter is sent to the API, it will break and the information returned to the user may not be as clear as if there was a validation in place and a custo message to return about the parameter errors. many APIs break with a `400` error for bad request, and so son.
thoughts?

Reply:
here are two runs for the `download-files` actions, 
in the one validating presence we get

![image](https://user-images.githubusercontent.com/25184117/131624065-a0739c7a-b238-4ca2-b4d2-611d7b71f867.png)


if we bypass presence with referencing a value from a previous steps, the pCloud returns a object, but the user still has to browse the returned data for the errors.

![image](https://user-images.githubusercontent.com/25184117/131624166-ca46609d-f111-4006-9b11-e0beeec97ca0.png)

i didn't tried for domainLocaiton we are going to have that soon via the `auths`variable.